samza-core/src/main/java/org/apache/samza/zk/ZkLeaderElector.java

```diff
@@ -84,6 +84,7 @@ private String getHostName() {
 
   @Override
   public void tryBecomeLeader(LeaderElectorListener leaderElectorListener) {
+    zkUtils.makeSurePersistentPathsExists(new String[]{keyBuilder.getProcessorsPath()});
```

Comment:
>  I guess he wanted to create all the zk nodes needed for ZkCoordinator together there. However, this doesn't work for coordinator service which has ZkLeaderElector but not ZkController.

Yes. CoordinationService usage should be independent of components in the JobCoordinator abstraction. 

> . So I added this logic back here. 

I don't think that is the right solution. We should not pollute the leader election process with Zk node setup. 

> This way they can be used independently. What do you think?

Yep. That's sounds right.